scripts/lighthouse/lighthouse.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable eslint-comments/no-unlimited-disable */
+/* eslint-disable */
+const lighthouse = require('lighthouse');
+const chromeLauncher = require('chrome-launcher');
+const table = require('markdown-table');
+
+const ROUTES_MAPPINGS = require('./pages');
+const checkLighthouseScore = require('./lh-check-scores');
+const config = require('./lh-config');
+
+const toMarkdown = (results = {}) =>
+  table([['Metric', 'Value']].concat(Object.entries(results)));
+
+const generateOutput = (scores, url) => {
+  console.log('URL', url);
+
+  if (scores.successes) {
+    console.log('\n===== LIGHTHOUSE PASS! =====\n');
+    console.log(toMarkdown(scores.successes));
+    console.log('\n============================\n');
+  } else {
+    console.log('\n===== LIGHTHOUSE FAILED! =====\n');
+    console.log(toMarkdown(scores.errors));
+    console.log('\n============================\n');
+  }
+};
+
+function launchChromeAndRunLighthouse(url, opts, conf) {
+  return chromeLauncher
+    .launch({ chromeFlags: opts.chromeFlags })
+    .then(chrome => {
+      opts.port = chrome.port;
+      return lighthouse(url, opts, conf).then(async results => {
```

Comment:
Does this handle the error case? It seems to only deal with the fulfillment case

Reply:
Could you clarify what you mean by error case?

Reply:
It does not have `catch` block anywhere. If this fails will fail with `unhandled promise error` which is not ideal

Reply:
That is true! Can blame the docs from lighthouse on that for now error handling 😂